fdbserver/CommitProxyServer.actor.cpp

```diff
@@ -1962,102 +2129,8 @@ ACTOR Future<Void> commitProxyServerCore(CommitProxyInterface proxy,
-						                                   SERVER_KNOBS->BUGGIFIED_ROW_LIMIT,
-						                                   SERVER_KNOBS->APPLY_MUTATION_BYTES)
-						                       .get();
-						if (!data.size())
-							break;
-						((KeyRangeRef&)txnKeys) = KeyRangeRef(keyAfter(data.back().key, txnKeys.arena()), txnKeys.end);
-
-						MutationsVec mutations;
-						std::vector<std::pair<MapPair<Key, ServerCacheInfo>, int>> keyInfoData;
-						vector<UID> src, dest;
-						ServerCacheInfo info;
-						for (auto& kv : data) {
-							if (kv.key.startsWith(keyServersPrefix)) {
-								KeyRef k = kv.key.removePrefix(keyServersPrefix);
-								if (k != allKeys.end) {
-									decodeKeyServersValue(tag_uid, kv.value, src, dest);
-									info.tags.clear();
-									info.src_info.clear();
-									info.dest_info.clear();
-									for (const auto& id : src) {
-										auto storageInfo =
-										    getStorageInfo(id, &commitData.storageCache, commitData.txnStateStore);
-										ASSERT(storageInfo->tag != invalidTag);
-										info.tags.push_back(storageInfo->tag);
-										info.src_info.push_back(storageInfo);
-									}
-									for (const auto& id : dest) {
-										auto storageInfo =
-										    getStorageInfo(id, &commitData.storageCache, commitData.txnStateStore);
-										ASSERT(storageInfo->tag != invalidTag);
-										info.tags.push_back(storageInfo->tag);
-										info.dest_info.push_back(storageInfo);
-									}
-									uniquify(info.tags);
-									keyInfoData.emplace_back(MapPair<Key, ServerCacheInfo>(k, info), 1);
-								}
-							} else {
-								mutations.emplace_back(mutations.arena(), MutationRef::SetValue, kv.key, kv.value);
-							}
-						}
-
-						// insert keyTag data separately from metadata mutations so that we can do one bulk insert which
-						// avoids a lot of map lookups.
-						commitData.keyInfo.rawInsert(keyInfoData);
-
-						Arena arena;
-						bool confChanges;
-						applyMetadataMutations(SpanID(),
-						                       commitData,
-						                       arena,
-						                       Reference<ILogSystem>(),
-						                       mutations,
-						                       /* pToCommit= */ nullptr,
-						                       confChanges,
-						                       /* popVersion= */ 0,
-						                       /* initialCommit= */ true);
-					}
-
-					auto lockedKey = commitData.txnStateStore->readValue(databaseLockedKey).get();
-					commitData.locked = lockedKey.present() && lockedKey.get().size();
-					commitData.metadataVersion = commitData.txnStateStore->readValue(metadataVersionKey).get();
-
-					commitData.txnStateStore->enableSnapshot();
-				}
-			}
-			addActor.send(broadcastTxnRequest(req, SERVER_KNOBS->TXN_STATE_SEND_AMOUNT, true));
-			wait(yield());
+		when(TxnStateRequest request = waitNext(proxy.txnState.getFuture())) {
+			wait(processTransactionStateRequestPart(&transactionStateResolveContext, request));
```

Comment:
It seems the original code is non-blocking, so:

```suggestion
			addActor.send(processTransactionStateRequestPart(&transactionStateResolveContext, request));
```